### ISSUE TITLE
Some Jamendo tracks do not allow downloads

### ISSUE THREAD

**Issue by stacimc** [WordPress/openverse#3499]
## Description

Some Jamendo records with CC licenses nevertheless do not allow audio downloads from the Jamendo site. 

Thanks @fcoveram for reporting

## Reproduction

1. Check out this audio result in Openverse: https://openverse.org/audio/74d3a2a9-88ce-4232-b165-2ac8612dd0c5?q=jazz
2. Click the `Get this audio` button to be taken to the record's page on Jamendo
3. On the Jamendo page, click the `Free Download` button
5. Observe that the download button is disabled with the message "Download not available: This artist has disabled the download of this track on Jamendo Music."

### Audio result on Openverse

<img width="872" alt="Screenshot 2023-12-08 at 9 34 58 AM" src="https://github.com/WordPress/openverse/assets/63313398/33e2debc-e462-4d3b-b369-b33d3997e584">

### Audio result on Jamendo
<img width="889" alt="Screenshot 2023-12-08 at 9 35 42 AM" src="https://github.com/WordPress/openverse/assets/63313398/70636c40-286d-47a3-b563-6dab21417083">

### Jamendo download page with downloads disabled
<img width="911" alt="Screenshot 2023-12-08 at 9 36 17 AM" src="https://github.com/WordPress/openverse/assets/63313398/efc75093-27f3-4318-83f0-2f62ff8daab9">

## Additional context

Tracks returned by the Jamendo API include an `audiodownload_allowed` field, separate from the license, which is not currrently used by Openverse. For some records with CC licenses, this field is nonetheless `False`.

<details>
  <summary>Here is an example API response from Jamendo for a record that has a CC license, but `audiodownload_allowed` is False</summary>
  
  
```
{
   "id":"1687",
   "name":"Rioting",
   "duration":235,
   "artist_id":"272",
   "artist_name":"Chroma",
   "artist_idstr":"chroma",
   "album_name":"first",
   "album_id":"248",
   "license_ccurl":"http://creativecommons.org/licenses/by-nc-sa/2.0/fr/",
   "position":1,
   "releasedate":"2005-07-08",
   "album_image":"[https://usercontent.jamendo.com?type=album&id=248&width=300&trackid=1687](https://usercontent.jamendo.com/?type=album&id=248&width=300&trackid=1687)",
   "audio":"https://prod-1.storage.jamendo.com/?trackid=1687&format=mp32&from=YG3K4BmvxKBdtt5swAthcw%3D%3D%7CnQCkm%2FibREubzaVo%2FkDYPw%3D%3D",
   "audiodownload":"https://prod-1.storage.jamendo.com/download/track/1687/mp32/",
   "prourl":"",
   "shorturl":"https://jamen.do/t/1687",
   "shareurl":"https://www.jamendo.com/track/1687",
   "lyrics":"",
   "audiodownload_allowed":false,
   "image":"[https://usercontent.jamendo.com?type=album&id=248&width=300&trackid=1687](https://usercontent.jamendo.com/?type=album&id=248&width=300&trackid=1687)",
   "musicinfo":{
      "vocalinstrumental":"instrumental",
      "gender":"neutral",
      "acousticelectric":"electric",
      "speed":"medium",
      "tags":{
         "genres":[
            "electronic"
         ],
         "instruments":[
            
         ],
         "vartags":[
            "neutral"
         ]
      }
   },
   "licenses":{
      "ccnc":"true",
      "ccnd":"false",
      "ccsa":"true",
      "prolicensing":"false",
      "probackground":"false",
      "cc":"true"
   },
   "stats":{
      "rate_downloads_total":462,
      "rate_listened_total":3003,
      "playlisted":13,
      "favorited":9,
      "likes":0,
      "dislikes":0,
      "avgnote":0,
      "notes":0
   }
}
```
  
</details>

We could update the Jamendo DAG to **exclude** records with `audiodownload_allowed` disabled. In order to backfill this change to already ingested records, a multi-step plan like the following could be used:

1. First, merge a change to the Jamendo DAG to write the `audiodownload_allowed` field in the `meta_data` column for _all_ records.
2. Run the Jamendo DAG, which is a non-dated DAG. This should write that field for all records. At this point, we'll also be able to see how many records meet this criteria.
3. Use the `delete_records` DAG to select all Jamendo records with `audiodownload_allowed` set to False. This DAG deletes those records from the audio table, but stores a copy in the `deleted_audio` table so they can be restored if we want.
4. Update the Jamendo DAG to discard records with `audiodownload_allowed` set to False, so no new records of this type are ingested.

**Comment by stacimc** [comment]
@WordPress/openverse-maintainers What do you think? Should the records be removed, or is there a better solution, perhaps by displaying additional context somewhere on our own result page?

These records appear to have legitimate CC licenses, but they cannot be downloaded. The user experience is particularly confusing because the button on Openverse's single audio result page reads "Get this audio".

**Comment by krysal** [comment]
Proven the records are still under CC-licensed, my opinion is that we shouldn't remove them. Openverse does not guarantee the original file is fully available to users (see the Freesound case), and the main goal is discoverability. It's important to make all the CC works available out there on the web known to the public, plus this behavior is probably common for works with an NC or ND license, where I assume each author would prefer users to request permissions individually.

I reiterate that this is my position if it is found that they still have a CC license, otherwise they can be deleted.

**Comment by stacimc** [comment]
It would be helpful to have a better understanding of how Jamendo is treating these. In the Freesound case the track _can_ be downloaded, you just have to log into Freesound.

Here, the free download button is disabled entirely. I tried signing up for Jamendo and was still unable to download the track. If you click the other button on the download page (`Get a license on Jamendo licensing`), you get [this page](https://licensing.jamendo.com/en/track/1106485/moonlight-jazz):

<img width="1082" alt="Screenshot 2023-12-08 at 12 04 26 PM" src="https://github.com/WordPress/openverse/assets/63313398/77faafbd-2b8a-4e94-91b6-57d5d447aadb">

A couple things:
* Note the text "or subscribe for unlimited access to all tracks for commercial online use." I'm not sure what the "subscribe" option here refers to, since I don't see any changes on this page when I'm logged in vs not logged in. Maybe there's an option I'm not seeing?
* From this page, there **is** a download button for a "low quality" version "for testing purposes only".
* [This](https://licensing.jamendo.com/en/pricing) is the page that's linked when you click "Learn more". It describes only paid tiers of Jamendo licenses.

There are zero mentions of the CC license on the download page. However, the icons _are_ still visible on the detail page, right above another link to the Jamendo licensing page:

<img width="1582" alt="Screenshot 2023-12-08 at 12 12 13 PM" src="https://github.com/WordPress/openverse/assets/63313398/96a918c3-e121-4939-9bc3-d90e5abf652c">


It's really unclear to me what's going on here. The API and the detail page both clearly indicate a CC license, but the messaging across all pages _also_ indicates that paid Jamendo license is necessary and the one available download is for "testing purposes only". But to @krysal's point, Openverse only guarantees that its records are CC-licensed, so I agree there's a strong case for leaving them in. I do think this might result in a frustrating experience for a user, however. It would be nice to have some way of filtering these out.

For contrast, here's the download page for a Jamendo record with `audiodownload_allowed` enabled -- note the CC license is reiterated here:

<img width="1082" alt="Screenshot 2023-12-08 at 12 14 12 PM" src="https://github.com/WordPress/openverse/assets/63313398/1eda7e21-a529-4348-a056-2696d327d7d1">

**Comment by fcoveram** [comment]
Interesting case.

I vote for removing these kinds of media items. The intention of assigning a CC license is to allow **the use** of work under _certain conditions_. However, it is true that Openverse stands for surfacing work under CC licenses to make them findable, but if a service sets a barrier to start using it, then there is a conceptual conflict where Openverse breaks down the expectation of using the media work.

So far, the main advice Openverse holds is asking users to check the license on the source site, but not to check if they can "use something" after clicking on "get this media."

**Comment by dhruvkb** [comment]
I personally lean towards keeping these tracks in our datastore (so that we are not gatekeeping licensed works from being indexed or not) but not surfacing them high in search results (so that our search results remain accessible and reusable). We could

- store this field in the audio track's metadata
- penalise the tracks with `audiodownload_allowed=False` in ES
- show a disclaimer for such results in the single result page

To be honest, I wouldn't even mind them being removed from the catalog as they are CC-licensed legally but not abiding by the free-sharing spirit of the licenses, but that puts us in the position of choosing what CC-licensed works we index and which we don't. That should not be something Openverse does.

**Comment by stacimc** [comment]
> I also presume we could easily recover these works through backfilling if we wanted to reintroduce them to the catalogue (for example, if discussions with Jamendo about their intentions with this give us a better understanding of how to present them in search).

Definitely! Since Jamendo is non-dated, if we want to reintroduce them to the catalog we'd just need to make a simple code change to stop filtering them at ingestion, and then run the DAG once (and do a data refresh, of course).

I really like @dhruvkb's idea to push these results down in search as an alternative, but it would involve significantly more work (including frontend work to add some sort of disclaimer on those individual result pages).

**Comment by krysal** [comment]
I want to emphasize what we mean by **using** here, as far as I can tell, the track in the issue can be played in full on Jamendo (and from Openverse as well). The users may not be able to download it but they can play it in Jamendo as many times as they want. The license also doesn't allow for remixes or adaptations. I don't get why we should make a downloadable file a prerequisite to include works in Openverse, as I said before, I think this will be a common experience for works with this licenses (NC, ND).

That's why I strongly prefer we add a `download_allowed` field and make it a filter instead of deleting them.

**Comment by fcoveram** [comment]
> If Jamendo is a big name that is misrepresenting their usage of CC to get the benefit of the perception of open access and open sharing but not actually following that ethos, it's more of a CC question than an Openverse one.

+ 100 to this.

> The license also doesn't allow for remixes or adaptations. I don't get why we should make a downloadable file a prerequisite to include works in Openverse, as I said before, I think this will be a common experience for works with this licenses (NC, ND).

[CC BY-NC-ND](https://creativecommons.org/licenses/by-nc-nd/4.0/) say that (quote) "You are free to: Share — copy and redistribute the material in any medium or format." Therefore you should be access to the file.

> I really like @dhruvkb's idea to push these results down in search as an alternative, but it would involve significantly more work (including frontend work to add some sort of disclaimer on those individual result pages).

This could also be a good idea, and so far, I envision a flow similar to surfacing sensitive content. The results have this filtered disabled by default, and once you click on something like "Show results that can not be downloaded" (the terrible copy is TBD indeed), you see them in the results. Then, the medial details page's CTA changes its label to something like "See the [media]"

**Comment by krysal** [comment]
Right! @fcoveram made me rethink, and now I support the idea of penalizing these works, but I'm not 100% convinced that a downloadable file should be a prerequisite to be in Openverse (for NC & ND licenses). "You" as a user are free to copy and redistribute, but is the creator obligated to share the file? Does being online free to play count? That is more of a question for CC folks, as mentioned before. There is also the possibility that the author may prefer people to contact them directly to obtain the file.

**Comment by AetherUnbound** [comment]
Thanks everyone for sharing your perspectives here! I agree that removing these records until we're able to handle them more appropriately (either with penalization, filters, or altered CTA on the details page) is the best way to go to prevent reputational harm to Openverse.

It seems that our immediate next step is to alter the Jamendo DAG to start capturing this information, does that seem right @stacimc? Would it be much work to add that to the `metadata` field?
